Sources/BraintreeCore/RepeatingTimer.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+
+final class RepeatingTimer {
+    
+    /// Amount of time, in seconds.
+    private let timeInterval: Int
+    
+    init(timeInterval: Int) {
```

Comment:
Since we've separate out this logic can we remove the `timeInterval` property from `BTAnalyticsService` and set the default in this class? That way the timer logic is fully contained here and we are only instantiating it in `BTAnalyticsService`.

Reply:
Yeh, we can just hard code it to 15 in the RepeatingTimer class. BTAnalyticsService doesn't need to change the interval

Reply:
@richherrera wanted to make sure you saw this

Reply:
I am not sure that I agree with this. `BTAnalyticsService` only responsibility should be sending events, it should not care about how often we send events at all. On the other hand the `RepeatingTimer` should control all time related logic including the interval of the timer. I don't think `BTAnalyticsService` needs to care how often events are sent only that it's sending events.

Reply:
Sure, I completely agree with you. I was thinking more about making this class usable in multiple places if needed, considering that in some other context, 15 seconds might be too short. But I'm happy to make the change. Do you think it would be good to put it in the initializer?